### PR TITLE
Add table azure_lb. Closes #182

### LINKED ISSUE #182: Add table azure_lb
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by c0d3r-arnab**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
arnab@turbotindias-MacBook-Pro azure-test % ./tint.js azure_lb
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/azure_lb []

PRETEST: tests/azure_lb

TEST: tests/azure_lb
Running terraform
azurerm_resource_group.named_test_resource: Creating...
azurerm_resource_group.named_test_resource: Creation complete after 1s [id=/subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest81844]
azurerm_lb.named_test_resource: Creating...
azurerm_lb.named_test_resource: Creation complete after 6s [id=/subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest81844/providers/Microsoft.Network/loadBalancers/turbottest81844]

Warning: Version constraints inside provider configuration blocks are deprecated

  on variables.tf line 21, in provider "azurerm":
  21:   version         = "=1.36.0"

Terraform 0.13 and earlier allowed provider version constraints inside the
provider configuration block, but that is now deprecated and will be removed
in a future version of Terraform. To silence this warning, move the provider
version constraint into the required_providers block.


Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

region = "westeurope"
resource_aka = "azure:///subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest81844/providers/Microsoft.Network/loadBalancers/turbottest81844"
resource_aka_lower = "azure:///subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourcegroups/turbottest81844/providers/microsoft.network/loadbalancers/turbottest81844"
resource_id = "/subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest81844/providers/Microsoft.Network/loadBalancers/turbottest81844"
resource_name = "turbottest81844"
subscription_id = "f2ecd657-84a7-4445-9b2a-280aa402eb9f"

Running SQL query: test-get-query.sql
[
  {
    "id": "/subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest81844/providers/Microsoft.Network/loadBalancers/turbottest81844",
    "name": "turbottest81844",
    "region": "westeurope",
    "resource_group": "turbottest81844",
    "subscription_id": "f2ecd657-84a7-4445-9b2a-280aa402eb9f",
    "tags": {
      "name": "turbottest81844"
    },
    "type": "Microsoft.Network/loadBalancers"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "id": "/subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest81844/providers/Microsoft.Network/loadBalancers/turbottest81844",
    "name": "turbottest81844"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "azure:///subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourceGroups/turbottest81844/providers/Microsoft.Network/loadBalancers/turbottest81844",
      "azure:///subscriptions/f2ecd657-84a7-4445-9b2a-280aa402eb9f/resourcegroups/turbottest81844/providers/microsoft.network/loadbalancers/turbottest81844"
    ],
    "name": "turbottest81844",
    "tags": {
      "name": "turbottest81844"
    },
    "title": "turbottest81844"
  }
]
✔ PASSED

POSTTEST: tests/azure_lb

TEARDOWN: tests/azure_lb

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
arnab@turbotindias-MacBook-Pro azure-test % steampipe query
Welcome to Steampipe v0.7.1
For more information, type .help
> select
  name,
  type,
  provisioning_state,
  etag,
  region
from
  azure_lb;
+------------------------------------+---------------------------------+--------------------+------------------------------------------+------------+
| name                               | type                            | provisioning_state | etag                                     | region     |
+------------------------------------+---------------------------------+--------------------+------------------------------------------+------------+
| turbot-test-20200125-create-update | Microsoft.Network/loadBalancers | Succeeded          | W/"62428496-51aa-4bb1-b25d-6562688615ed" | westeurope |
| test_load_balancer                 | Microsoft.Network/loadBalancers | Succeeded          | W/"4db96ea8-2253-4b1e-9a2d-987ae4149a7b" | eastus     |
+------------------------------------+---------------------------------+--------------------+------------------------------------------+------------+


 select
  name,
  provisioning_state
from
  azure_lb
where
  provisioning_state = 'Failed'
+------+--------------------+
| name | provisioning_state |
+------+--------------------+
+------+--------------------+
```
</details>

### CODE REVIEW

File: azure-test/tests/azure_lb/test-get-query.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+select name, id, type, region, resource_group, subscription_id, tags
+from azure.azure_lb
+where name = '{{resourceName}}' and resource_group = '{{resourceName}}'
```
Comment (bigdatasourav) [inline azure-test/tests/azure_lb/test-get-query.sql:3]:
```suggestion
where name = '{{resourceName}}' and resource_group = '{{resourceName}}';
```

File: azure-test/tests/azure_lb/test-get-expected.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[
+  {
+    "id": "{{ output.resource_id.value }}",
+    "name": "{{resourceName}}",
```
Comment (bigdatasourav) [inline azure-test/tests/azure_lb/test-get-expected.json:4]:
```suggestion
    "name": "{{ resourceName }}",
```

File: azure-test/tests/azure_lb/test-get-expected.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    "region": "{{ output.region.value }}",
+    "resource_group": "{{resourceName}}",
```
Comment (bigdatasourav) [inline azure-test/tests/azure_lb/test-get-expected.json:6]:
```suggestion
    "resource_group": "{{ resourceName }}",
```

File: azure-test/tests/azure_lb/test-list-expected.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+[
+  {
+    "id": "{{ output.resource_id.value }}",
+    "name": "{{resourceName}}"
```
Comment (bigdatasourav) [inline azure-test/tests/azure_lb/test-list-expected.json:4]:
```suggestion
    "name": "{{ resourceName }}"
```

File: azure-test/tests/azure_lb/test-list-query.sql [hunk 5]
```diff
@@ -0,0 +1,3 @@
+select name, id
+from azure.azure_lb
+where name = '{{resourceName}}'
```
Comment (bigdatasourav) [inline azure-test/tests/azure_lb/test-list-query.sql:3]:
```suggestion
where name = '{{ resourceName }}';
```

File: azure-test/tests/azure_lb/test-not-found-query.sql [hunk 6]
```diff
@@ -0,0 +1,3 @@
+select name, id, type, region
+from azure.azure_lb
+where name = 'dummy-test794612891' and resource_group = '{{resourceName}}'
```
Comment (bigdatasourav) [inline azure-test/tests/azure_lb/test-not-found-query.sql:3]:
```suggestion
where name = 'dummy-test{{ resourceName }}' and resource_group = '{{ resourceName }}';
```

File: azure-test/tests/azure_lb/variables.tf [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+variable "azure_environment" {
+  type        = string
+  default     = "public"
+  description = "Azure environment used for the test."
+}
+
+variable "azure_subscription" {
+  type        = string
+  default     = "f2ecd657-84a7-4445-9b2a-280aa402eb9f"
```
Comment (bigdatasourav) [inline azure-test/tests/azure_lb/variables.tf:15]:
```suggestion
  default     = "3510ae4d-530b-497d-8f30-53c0616fc6c1"
```

Comment (bigdatasourav) [reply azure-test/tests/azure_lb/variables.tf:15]: we will not provide the actual subscription id.

File: azure/table_azure_lb.go [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+
+	"github.com/Azure/azure-sdk-for-go/profiles/2020-09-01/monitor/mgmt/insights"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-02-01/network"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+)
+
+//// TABLE DEFINITION ////
```
Comment (bigdatasourav) [inline azure/table_azure_lb.go:14]:
```suggestion
//// TABLE DEFINITION
```

File: azure/table_azure_lb.go [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+			},
+			{
+				Name:        "id",
+				Description: "Resource ID.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromGo(),
+			},
+			{
+				Name:        "provisioning_state",
+				Description: "The provisioning state of the load balancer resource. Possible values include: 'ProvisioningStateSucceeded', 'ProvisioningStateUpdating', 'ProvisioningStateDeleting', 'ProvisioningStateFailed'",
```
Comment (bigdatasourav) [inline azure/table_azure_lb.go:42]:
```suggestion
				Description: "The provisioning state of the load balancer resource. Possible values include: 'Succeeded', 'Updating', 'Deleting', 'Failed'.",
```

File: azure/table_azure_lb.go [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+			Hydrate:           getLoadBalancer,
+			ShouldIgnoreError: isNotFoundError([]string{"ResourceNotFound", "ResourceGroupNotFound", "404"}),
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listLoadBalancers,
+		},
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Description: "Resource name.",
```
Comment (bigdatasourav) [inline azure/table_azure_lb.go:31]:
```suggestion
				Description: "The resource name.",
```

File: azure/table_azure_lb.go [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "id",
+				Description: "Resource ID.",
```
Comment (bigdatasourav) [inline azure/table_azure_lb.go:36]:
```suggestion
				Description: "The resource ID.",
```

File: azure/table_azure_lb.go [hunk 12]
```diff
@@ -0,0 +1,276 @@
...
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromGo(),
+			},
+			{
+				Name:        "provisioning_state",
+				Description: "The provisioning state of the load balancer resource. Possible values include: 'ProvisioningStateSucceeded', 'ProvisioningStateUpdating', 'ProvisioningStateDeleting', 'ProvisioningStateFailed'",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("LoadBalancerPropertiesFormat.ProvisioningState").Transform(transform.ToString),
```
Comment (bigdatasourav) [inline azure/table_azure_lb.go:44]: Do we need Transform(transform.ToString) ?

File: azure/table_azure_lb.go [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+			},
+			{
+				Name:        "type",
+				Description: "Resource type.",
```
Comment (bigdatasourav) [inline azure/table_azure_lb.go:48]:
```suggestion
				Description: "The resource type.",
```

File: azure/table_azure_lb.go [hunk 14]
```diff
@@ -0,0 +1,276 @@
...
+			},
+			{
+				Name:        "resource_guid",
+				Description: "The resource GUID property of the load balancer resource.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("LoadBalancerPropertiesFormat.ResourceGUID"),
+			},
+			{
+				Name:        "sku_name",
+				Description: "Name of a load balancer SKU. Possible values include: 'LoadBalancerSkuNameBasic', 'LoadBalancerSkuNameStandard', 'LoadBalancerSkuNameGateway'.",
```
Comment (bigdatasourav) [inline azure/table_azure_lb.go:76]:
```suggestion
				Description: "Name of the load balancer SKU. Possible values include: 'Basic', 'Standard', 'Gateway'.",
```

File: azure/table_azure_lb.go [hunk 15]
```diff
@@ -0,0 +1,276 @@
...
+				Name:        "subscription_id",
+				Description: ColumnDescriptionSubscription,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ID").Transform(idToSubscriptionID),
+			},
+		},
+	}
+}
+
+//// FETCH FUNCTIONS ////
```
Comment (bigdatasourav) [inline azure/table_azure_lb.go:178]:
```suggestion
//// LIST FUNCTION
```

File: azure/table_azure_lb.go [hunk 16]
```diff
@@ -0,0 +1,276 @@
...
+		for _, loadBalancer := range result.Values() {
+			d.StreamListItem(ctx, loadBalancer)
+		}
+		result.NextWithContext(context.Background())
+		pagesLeft = result.NotDone()
+	}
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS ////
```
Comment (bigdatasourav) [inline azure/table_azure_lb.go:206]:
```suggestion
//// HYDRATE FUNCTIONS
```

File: azure/table_azure_lb.go [hunk 17]
```diff
@@ -0,0 +1,276 @@
...
+
+func getLoadBalancer(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getLoadBalancer")
+
+	name := d.KeyColumnQuals["name"].GetStringValue()
+	resourceGroup := d.KeyColumnQuals["resource_group"].GetStringValue()
```
Comment (bigdatasourav) [inline azure/table_azure_lb.go:212]: There should be a empty check for these two values

File: docs/tables/azure_lb.md [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+  name,
+  type,
+  provisioning_state,
+  etag,
+  region
+from
+  azure_lb;
+```
+
+### List load balancer with failed provisioning state
```
Comment (bigdatasourav) [inline docs/tables/azure_lb.md:20]:
```suggestion
### List failed load balancers
```

**Review by bigdatasourav — CHANGES_REQUESTED** [review]
@c0d3r-arnab Please check the suggested changes

File: azure/table_azure_lb.go (hunk 12 above)
Comment (c0d3r-arnab) [reply azure/table_azure_lb.go:44]: It will come as `null` otherwise

Comment (c0d3r-arnab) [reply azure/table_azure_lb.go:44]: I understood the comment wrong. I have made the changes

File: azure-test/tests/azure_lb/test-get-query.sql [hunk 19]
```diff
@@ -0,0 +1,3 @@
+select name, id, type, region, resource_group, subscription_id, tags
+from azure.azure_lb
+where name = '{{ resourceName }}' and resource_group = '{{ resourceName }}'
```
Comment (bigdatasourav) [inline azure-test/tests/azure_lb/test-get-query.sql:3]:
```suggestion
where name = '{{ resourceName }}' and resource_group = '{{ resourceName }}';
```

File: azure-test/tests/azure_lb/test-list-query.sql [hunk 20]
```diff
@@ -0,0 +1,3 @@
+select name, id
+from azure.azure_lb
+where name = '{{ resourceName }}'
```
Comment (bigdatasourav) [inline azure-test/tests/azure_lb/test-list-query.sql:3]:
```suggestion
where name = '{{ resourceName }}';
```

File: azure-test/tests/azure_lb/test-not-found-query.sql [hunk 21]
```diff
@@ -0,0 +1,3 @@
+select name, id, type, region
+from azure.azure_lb
+where name = 'dummy-test{{ resourceName }}' and resource_group = '{{ resourceName }}'
```
Comment (bigdatasourav) [inline azure-test/tests/azure_lb/test-not-found-query.sql:3]:
```suggestion
where name = 'dummy-test{{ resourceName }}' and resource_group = '{{ resourceName }}';
```

File: azure-test/tests/azure_lb/test-turbot-query.sql [hunk 22]
```diff
@@ -0,0 +1,3 @@
+select name, akas, tags, title
+from azure.azure_lb
+where name = '{{ resourceName }}' and resource_group = '{{ resourceName }}'
```
Comment (bigdatasourav) [inline azure-test/tests/azure_lb/test-turbot-query.sql:3]:
```suggestion
where name = '{{ resourceName }}' and resource_group = '{{ resourceName }}';
```

File: azure/table_azure_lb.go [hunk 23]
```diff
@@ -0,0 +1,281 @@
...
+			},
+			{
+				Name:        "id",
+				Description: "The resource ID.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromGo(),
+			},
+			{
+				Name:        "provisioning_state",
+				Description: "The provisioning state of the load balancer resource. Possible values include: 'Succeeded', 'Updating', 'Deleting', 'Failed'",
```
Comment (bigdatasourav) [inline azure/table_azure_lb.go:42]:
```suggestion
				Description: "The provisioning state of the load balancer resource. Possible values include: 'Succeeded', 'Updating', 'Deleting', 'Failed'.",
```

File: azure/table_azure_lb.go [hunk 24]
```diff
@@ -0,0 +1,281 @@
...
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "frontend_ip_configurations",
+				Description: "Object representing the frontend IPs to be used for the load balancer.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("LoadBalancerPropertiesFormat.FrontendIPConfigurations"),
+			},
+			{
+				Name:        "inbound_nat_pools",
```
Comment (bigdatasourav) [inline azure/table_azure_lb.go:106]: why we have `inbound_nat_pools` two times?

File: docs/tables/azure_lb.md [hunk 25]
```diff
@@ -0,0 +1,30 @@
+# Table: azure_lb
+
+An Azure load balancer is a Layer-4 (TCP, UDP) load balancer that provides high availability by distributing incoming traffic among healthy VMs.
```
Comment (bigdatasourav) [inline docs/tables/azure_lb.md:3]:
```suggestion
Azure Load Balancer operates at layer 4 of the Open Systems Interconnection (OSI) model. It's the single point of contact for clients. Load balancer distributes inbound flows that arrive at the load balancer's front end to backend pool instances. These flows are according to configured load-balancing rules and health probes. The backend pool instances can be Azure Virtual Machines or instances in a virtual machine scale set.
```

File: docs/tables/azure_lb.md [hunk 26]
```diff
@@ -0,0 +1,30 @@
...
+### List failed load balancers
+
+```sql
+select
+  name,
+  provisioning_state
+from
+  azure_lb
+where
+  provisioning_state = 'Failed'
```
Comment (bigdatasourav) [inline docs/tables/azure_lb.md:29]:
```suggestion
  provisioning_state = 'Failed';
```

File: docs/tables/azure_lb.md [hunk 27]
```diff
@@ -0,0 +1,30 @@
...
+  region
+from
+  azure_lb;
+```
+
+### List failed load balancers
+
+```sql
+select
+  name,
```
Comment (bigdatasourav) [inline docs/tables/azure_lb.md:24]: Add id and type.

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**